src/TransportTests/ConfigureAzureServiceBusTransportInfrastructure.cs

```diff
@@ -39,7 +40,8 @@ string HexStringFromBytes(byte[] bytes)
                         return sb.ToString();
                     }
                 }
-            }
+            },
+            SubscriptionRuleFilterConvention = eventType => new CorrelationRuleFilter(eventType.FullName)
```

Comment:
Is the `Headers.EnclosedMessageTypes` value going to be propagated to the message's `CorrelationId` property?

Reply:
@SeanFeldman This is only a rebase of:

- https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/pull/884

From what I read correlation filter work on application specific message properties too?

Why should it be assigned to the `CorrelationId` property?

Reply:
> From what I read correlation filter work on application specific message properties too?

Correct.

> Why should it be assigned to the CorrelationId property?

You're setting a value (message type) through the constructor. That value by default will be compared against `CorrelationId` system property.